app/knights.py

```diff
@@ -0,0 +1,38 @@
+class Knights:
+    # This constructor is creating a knight (Knights class instance),
+    # which is ready for the battle
+    # (armoured and potion applied (if available)).
+    def __init__(
+            self,
+            name: str,
+            hp: int,
+            power: int,
+            armour: list,
+            weapon: dict,
+            potion: dict | None) -> None:
+        self.name = name
+        self.hp = hp
+        self.protection = 0
+        if armour:
+            for item in armour:
+                self.protection += item.get("protection")
+        extra_power = 0
+        if potion:
+            self.hp = hp + potion.get("effect").get("hp")
+            extra_power = potion.get("effect").get("power")
+            if potion.get("effect").get("protection"):
+                self.protection += potion.get("effect").get("protection")
+        self.power = power + extra_power + weapon.get("power")

```

Comment:
Add methods for this. Or even create another modules and classes for each object (armour, potion...)